semigroups.cabal

```diff
@@ -128,6 +136,9 @@ library
     if flag(bytestring)
       build-depends: bytestring >= 0.9 && < 1
 
+    if flag(bytestring-builder) && impl(ghc < 7.8)
```

Comment:
I suppose someone with GHC 7.8 or later *could* install an older version of `bytestring` manually, but in practice I've never seen `cabal-install` pick such a build plan.